components/new_post/post_component.tsx

```diff
@@ -0,0 +1,500 @@
+    isEmbedVisible?: boolean;
+    enableEmojiPicker?: boolean;
+    enablePostUsernameOverride?: boolean;
+    isReadOnly?: boolean;
+    pluginPostTypes?: {[postType: string]: PostPluginComponent};
+    channelIsArchived?: boolean;
+    isConsecutivePost?: boolean;
+    isLastPost?: boolean;
+    recentEmojis: Emoji[];
+    handleCardClick?: (post: Post) => void;
+    togglePostMenu?: (opened: boolean) => void;
+    channelName?: string;
+    displayName: string;
+    teamDisplayName?: string;
+    teamName?: string;
+    channelType?: string;
+    a11yIndex?: number;
+    isBot: boolean;
+    hasReplies?: boolean;
+    isFirstReply?: boolean;
+    previousPostIsComment?: boolean;
+    matches?: string[];
+    term?: string;
+    isMentionSearch?: boolean;
+    location: keyof typeof Locations;
+    actions: {
+        markPostAsUnread: (post: Post, location: string) => void;
+        emitShortcutReactToLastPostFrom: (emittedFrom: 'CENTER' | 'RHS_ROOT' | 'NO_WHERE') => void;
+        setActionsMenuInitialisationState: (viewed: Record<string, boolean>) => void;
+        selectPost: (post: Post) => void;
+        removePost: (post: Post) => void;
+        closeRightHandSide: () => void;
+        selectPostCard: (post: Post) => void;
+        setRhsExpanded: (rhsExpanded: boolean) => void;
+    };
+    timestampProps?: Partial<TimestampProps>;
+    shouldHighlight?: boolean;
+    isPostBeingEdited?: boolean;
+    isCollapsedThreadsEnabled?: boolean;
+    isMobileView: boolean;
+    canReply?: boolean;
+    replyCount?: number;
+};
+
+const PostComponent = (props: Props): JSX.Element => {
+    const isSearchResultItem = (props.matches && props.matches.length > 0) || props.isMentionSearch || (props.term && props.term.length > 0);
+    const postRef = useRef<HTMLDivElement>(null);
+    const postHeaderRef = useRef<HTMLDivElement>(null);
+    const [hover, setHover] = useState(false);
+    const [a11yActive, setA11y] = useState(false);
+    const [dropdownOpened, setDropdownOpened] = useState(false);
+    const [fileDropdownOpened, setFileDropdownOpened] = useState(false);
+    const [fadeOutHighlight, setFadeOutHighlight] = useState(false);
+    const [alt, setAlt] = useState(false);
+
+    useEffect(() => {
+        if (props.shouldHighlight) {
+            const timer = setTimeout(() => setFadeOutHighlight(true), Constants.PERMALINK_FADEOUT);
+            return () => {
+                clearInterval(timer);
+            };
+        }
+    }, [props.shouldHighlight]);
+
+    useEffect(() => {
+        if (postRef.current) {
+            postRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, handleA11yActivateEvent);
+            postRef.current.addEventListener(A11yCustomEventTypes.DEACTIVATE, handleA11yDeactivateEvent);
+            addKeyboardListeners();
+        }
+        if (a11yActive) {
+            postRef.current?.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
+        }
+    }, [hover, postRef.current]);
```

Comment:
ref should not be used as dependencies, because they are not tracked for updates.